web/server/routers/post.ts

```diff
@@ -126,62 +134,53 @@ export const postRouter = router({
       z.object({
         email: z.string(),
         postOid: zodOidType,
-    })
-  ).mutation(async({input}) => {
-    const user = await findUserByEmail(input.email);
-    if (!user) {
-      throw new TRPCError({
-        message: "No user with given email exists.",
-        code: "BAD_REQUEST",
-      });
-    }
-    let email = "";
-    try {
-      const post = await getPost(input.postOid);
-      switch(post.type) {
-        case "fosterMove":
-          email = "foster@angelsrescue.org";
-          break;
-        case "return":
-          email = "returns@angelsrescue.org, foster@angelsrescue.org";
-          break;
-        case "temporary":
-          email = "tempfoster@angelsrescue.org";
-          break;
-        case "boarding":
-          email = "boardingadmin@angelsrescue.org"
-        case "shelter":
-        case "ownerSurrender":
-          email = "fosteroffer@angelsrescue.org";
-          break;
-      }
-      if (!email) {
+      })
+    )
+    .mutation(async ({ input }) => {
+      try {
+        const user = await findUserByEmail(input.email);
+        if (!user) {
+          throw new TRPCError({
+            message: "No user with given email exists.",
+            code: "BAD_REQUEST",
+          });
+        }
+      } catch (e) {
         throw new TRPCError({
-          message: "Post has invalid type.",
-          code: "BAD_REQUEST",
+          message: "Server unable to access user endpoint.",
+          code: "INTERNAL_SERVER_ERROR",
         });
       }
       try {
-        const info = await transporter.sendMail({
-          from: '"Angels Among Us Fostering Portal" <bitsofgood.aau@gmail.com>', 
-          to: email, 
-          subject: "Someone is ready to foster your dog!", 
-          text: "User has signed up to foster dog, a stray dog.", 
-        });
+        const post = await getPost(input.postOid);
+        const email = (fosterTypeEmails as any)[post.type] || "";
+        if (email.length < 2) {
+          throw new TRPCError({
+            message: "Post has invalid type.",
+            code: "BAD_REQUEST",
+          });
+        }
```

Comment:
This might be a bit unnecessary. The value of `2` also isn't exactly clear to me - avoid using magic numbers whenever possible. Perhaps you could remove this check since we are guaranteed by TypeScript that `email` is a value of `fosterTypeEmails`.

Reply:
This line was in case the type of the post was not one of the provided types in the email record, or potentially even empty. The value of email is `""` if the post type does not map to any of the post types listed in the record. I have changed it to `if (email == "")`. I get that it's an enum- but if the application evolves such that there are other potential values or if it's possible to not specify a type, I just incorporated the check just in case. Let me know if you want to keep it with the `email == ""` or not.

Reply:
I see. When an object is typed as a record with an enum (like the way we have it, `Record<FosterType, string>`), TypeScript will throw an error if any of the possible `FosterType` enum values are not included as well as if there are any extra fields. I don't believe the check is necessary since TypeScript has type guarantees, and if we make any changes that affect `FosterType`, our code won't compile.